From 27310f474957020b3083605083d86c93a812b42a Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Wed, 7 Aug 2024 05:15:39 +0000
Subject: [PATCH] [CMake] Exclude FoundationNetworking and
 _CFURLSessionInterface on WASI

Because networking is not a part of WASI Preview 1. We can add it back
when it is available.
---
 Sources/CMakeLists.txt | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/Sources/CMakeLists.txt b/Sources/CMakeLists.txt
index 0ee266a4..29b92440 100644
--- a/Sources/CMakeLists.txt
+++ b/Sources/CMakeLists.txt
@@ -14,10 +14,14 @@
 
 add_subdirectory(CoreFoundation)
 add_subdirectory(_CFXMLInterface)
-add_subdirectory(_CFURLSessionInterface)
+if(NOT CMAKE_SYSTEM_NAME STREQUAL "WASI")
+    add_subdirectory(_CFURLSessionInterface)
+endif()
 add_subdirectory(Foundation)
 add_subdirectory(FoundationXML)
-add_subdirectory(FoundationNetworking)
+if(NOT CMAKE_SYSTEM_NAME STREQUAL "WASI")
+    add_subdirectory(FoundationNetworking)
+endif()
 if(FOUNDATION_BUILD_TOOLS)
 	add_subdirectory(plutil)
 endif()
-- 
2.43.2

